UPGRADING.md

```diff
@@ -13,7 +13,126 @@ limitations under the License.
 
 # 3.0.0 Migration Guide
 
-TODO
+## New Required Dependencies
+
+Some of the previously optional dependencies are now *required* in `3.x` versions of the
+library, namely
+[google-cloud-bigquery-storage](https://pypi.org/project/google-cloud-bigquery-storage/)
+(minimum version `2.0.0`) and [pyarrow](https://pypi.org/project/pyarrow/) (minimum
+version `3.0.0`).
+
+The behavior of some of the package "extras" has thus also changed:
```

Comment:
Let's add a bullet that the `pandas` extra now requires the `db-dtypes` package.

Reply:
Will do.

BTW, I noticed that `db-types` is currently marked as Beta, which implies that that `pandas` extra is also just Beta, technically at least. Should we mention that as well or are we expected to promote `db-types` to stable before we release `v3`?

Reply:
This has been resolved. db-dtypes 1.0.0 has been released https://github.com/googleapis/python-db-dtypes-pandas/pull/94